todo_app/app.py

```diff
@@ -1,15 +1,47 @@
-from flask import Flask
-
+from flask import Flask, render_template, request, redirect, url_for
 from todo_app.flask_config import Config
 
+from todo_app.data.session_items import get_items, add_item, save_item, get_item, remove_item
+
 app = Flask(__name__)
 app.config.from_object(Config)
 
 
 @app.route('/')
 def index():
-    return 'Hello World!'
+    return render_template('index.html', todoList = get_items())
+
+@app.route('/new_todo', methods=['POST'])
+def add_item_from_form():
+    title = request.form['title']
+    add_item(title)
+    return redirect(url_for('index'))
+
+@app.route('/update_todo/<id>', methods=['POST'])
+def update_item(id):
+    ticked = False
+    try:
+        if request.form['completed'] == 'on':
+            ticked = True 
+    finally:
+        item = get_item(id)
+        if ticked:
+            item['status'] = 'Completed'
+        else:
+            item['status'] = 'Not Started'
+        save_item(item)
+        return redirect(url_for('index'))
```

Comment:
You can use dict.get('your key') as a safe way of getting a value out. Either specify a default value to be used when the key isn't found, or let it default to `None`.
None is falsy and strings like 'on' are truthy, meaning they will fail/pass the if condition as desired
i.e.:
```
    item = get_item(id)
    if request.form.get('completed'):
        item['status'] = 'Completed'
    else:
        item['status'] = 'Not Started'
    save_item(item)
    return redirect(url_for('index'))
```

Reply:
I didn't spot the None at get_item method in session_items.py:

next((item for item in items if item['id'] == int(id)), **None**)